happi/cli.py

```diff
@@ -97,223 +47,314 @@ def happi_cli(args):
-                        item_value = default
-                    else:
-                        logger.info('Not an optional field!')
-                        continue
-                try:
-                    info.enforce_value(item_value)
-                    valid_value = True
-                    kwargs[info.key] = item_value
-                except Exception:
-                    logger.info(f'Invalid value {item_value}')
-
-        device = client.create_device(container, **kwargs)
-        logger.info('Please confirm the following info is correct:')
-        device.show_info()
-        ok = input('y/N\n')
-        if 'y' in ok:
-            logger.info('Adding device')
-            device.save()
+            return
+        if value.replace('.', '').isnumeric():
+            logger.debug('Changed %s to float', value)
+            value = str(float(value))
+
+        if is_a_range(value):
+            start, stop = value.split(',')
+            start = float(start)
+            stop = float(stop)
+            is_range = True
+            if start < stop:
+                range_list = client.search_range(criteria, start, stop)
+            else:
+                logger.error('Invalid range, make sure start < stop')
+
+        # skip the criteria for range values
+        # it won't be a valid criteria for search_regex()
+        if is_range:
+            pass
         else:
-            logger.info('Aborting')
-    elif args.cmd == 'edit':
-        logger.debug('Starting edit block')
-        device = client.find_device(name=args.name)
-        is_invalid_field = False
-        for edit in args.edits:
-            field, value = edit.split('=', 1)
-            try:
-                getattr(device, field)
-                logger.info('Setting %s.%s = %s', args.name, field, value)
-                setattr(device, field, value)
-            except Exception as e:
-                is_invalid_field = True
-                logger.error('Could not edit %s.%s: %s', args.name, field, e)
-        if is_invalid_field:
-            sys.exit(1)
+            client_args[criteria] = fnmatch.translate(value)
+
+    regex_list = client.search_regex(**client_args)
+    results = regex_list + range_list
+
+    # find the repeated items
+    res_size = len(results)
+    repeated = []
+    for i in range(res_size):
+        k = i + 1
+        for j in range(k, res_size):
+            if results[i] == results[j] and results[i] not in repeated:
+                repeated.append(results[i])
+
+    # we only want to return the ones that have been repeated when
+    # they have been matched with both search_regex() & search_range()
+    if repeated:
+        final_results = repeated
+    elif regex_list and not is_range:
+        # only matched with search_regex()
+        final_results = regex_list
+    elif range_list and is_range:
+        # only matched with search_range()
+        final_results = range_list
+    else:
+        final_results = []
```

Comment:
I'd like to revisit all of this... Out of scope of this PR

Reply:
This was definitely a "tests pass moving on" moment for me 😬